qiita_ware/ebi.py

```diff
@@ -418,22 +430,40 @@ def _generate_spot_descriptor(self, design, platform):
         base_coord = ET.SubElement(read_spec, 'BASE_COORD')
         base_coord.text = '1'
 
-    def generate_experiment_xml(self):
+    def generate_experiment_xml(self, samples=None):
         """Generates the experiment XML file
 
+        Parameters
+        ----------
+        samples : list of str, optional
+            The list of samples to be included in the experiment xml
+
         Returns
         -------
         ET.Element
             Object with experiment XML values
         """
-        study_alias = self._get_study_alias()
+        study_accession = self.study.ebi_study_accession
+        if study_accession:
+            study_ref_dict = {'accession': study_accession}
+        else:
+            study_ref_dict = {'refname': self._get_study_alias()}
+
         experiment_set = ET.Element('EXPERIMENT_SET', {
             'xmlns:xsi': self.xmlns_xsi,
             "xsi:noNamespaceSchemaLocation": self.xsi_noNSL % "experiment"})
-        for sample_name, sample_prep in sorted(self.samples_prep.items()):
-            sample_alias = self._get_sample_alias(sample_name)
+
+        samples = samples if samples is not None else viewkeys(self.samples)
+
+        for sample_name in sorted(samples):
             experiment_alias = self._get_experiment_alias(sample_name)
             sample_prep = dict(self.samples_prep[sample_name])
+            if self._ebi_sample_accessions[sample_name]:
```

Comment:
It shouldn't. `samples` is a subset of the samples in the sample template, and `_ebi_sample_accessions` has an entry for each of the samples in the sample template.